### PR TITLE
fix: vertically center align the bounded text correctly when zoomed

### PR DESCRIPTION
**Pull request by ad1992**
Earlier: 
Excalidraw (27)

Now
Excalidraw (28)

Also fixes https://github.com/excalidraw/excalidraw/issues/4421, https://github.com/excalidraw/excalidraw/issues/4418
try https://1m7pt.csb.app/

### CODE REVIEW

**Comment by dwelle** [comment]
Nice. But, this also breaks the offset fix from previous:

excal-pr-sticky-calc-offset-bug

**Review by dwelle — APPROVED** [review]

**Comment by zsviczian** [comment]
This is getting better by the minute! 😀

I found two small glitches:

https://user-images.githubusercontent.com/14358394/146751885-45f36212-ff9c-4ae1-b6b0-a32dbc8befeb.mp4

**Comment by ad1992** [comment]
ok so Its somewhat better now but not completely fixed yet, there is some ~14px diff I am getting when offsets present, I have added a comment https://github.com/excalidraw/excalidraw/pull/4444/commits/81c76d121eeb039ded18adb210466778e4d0ba59
mostly with scrollable containers its not working well, I think we can merge this and will check more on this and improve it. 
The overflow bug has been there from beginning for text editor (until we started calculating max height of text editor in sticky notes PR) since we use position absolute but somehow with offsets only its not working very well but better now

File: src/element/textWysiwyg.tsx [hunk 1]
```diff
@@ -192,19 +195,35 @@ export const textWysiwyg = ({
...
       // Make sure text editor height doesn't go beyond viewport
-      const editorMaxHeight =
-        (appState.offsetTop + appState.height - viewportY) /
+      editorMaxHeight =
+        (appState.height -
+          viewportY -
+          // There is a ~14px difference which keeps on increasing
+          // with every zoom when offset present hence I am substracting it here
+          // However this is not the best fix and breaks in
+          // few scenarios
```
Comment (ad1992) [inline src/element/textWysiwyg.tsx:206]:
```suggestion
          // There is a ~14px difference which keeps on increasing
          // with every zoom step when offset present hence I am subtracting it here
          // However this is not the best fix and breaks in
          // few scenarios
```

**Review by dwelle — APPROVED** [review]

**Pull request merged**